File: backend/src/main/java/com/ll/hotel/domain/hotel/room/type/BedTypeNumber.java
package com.ll.hotel.domain.hotel.room.type;

import com.fasterxml.jackson.annotation.JsonProperty;
import jakarta.persistence.Embeddable;
import java.util.Map;

@Embeddable
public record BedTypeNumber(
        @JsonProperty("SINGLE")
        int bed_single,

        @JsonProperty("DOUBLE")
        int bed_double,

        @JsonProperty("QUEEN")
        int bed_queen,

        @JsonProperty("KING")
        int bed_king,

        @JsonProperty("TWIN")
        int bed_twin,

        @JsonProperty("TRIPLE")
        int bed_triple
) {
    public static BedTypeNumber fromJson(Map<String, Integer> bedTypeNumber) {
        return new BedTypeNumber(
                bedTypeNumber.getOrDefault("SINGLE", 0),
                bedTypeNumber.getOrDefault("DOUBLE", 0),
                bedTypeNumber.getOrDefault("QUEEN", 0),
                bedTypeNumber.getOrDefault("KING", 0),
                bedTypeNumber.getOrDefault("TWIN", 0),
                bedTypeNumber.getOrDefault("TRIPLE", 0)
        );
    }
}

File: backend/src/main/java/com/ll/hotel/domain/hotel/room/type/BedTypeNumber.kt
package com.ll.hotel.domain.hotel.room.type

import jakarta.persistence.Embeddable

@Embeddable
data class BedTypeNumber(
    val bedSingle: Int = 0,
    val bedDouble: Int = 0,
    val bedQueen: Int = 0,
    val bedKing: Int = 0,
    val bedTwin: Int = 0,
    val bedTriple: Int = 0
) {
    companion object {
        fun fromJson(bedTypeNumber: Map<String, Int>): BedTypeNumber {
            return BedTypeNumber(
                bedSingle = bedTypeNumber["SINGLE"] ?: 0,
                bedDouble = bedTypeNumber["DOUBLE"] ?: 0,
                bedQueen = bedTypeNumber["QUEEN"] ?: 0,
                bedKing = bedTypeNumber["KING"] ?: 0,
                bedTwin = bedTypeNumber["TWIN"] ?: 0,
                bedTriple = bedTypeNumber["TRIPLE"] ?: 0
            )
        }
    }
}